# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ObjectGroup.h
# Commit: 11fc8f557a2a
# Full Hash: 11fc8f557a2a7cf3c89e7c1396316aabd9725778
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 9bfa6023a553 -r 11fc8f557a2a js/src/vm/ObjectGroup.h
--- a/js/src/vm/ObjectGroup.h	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/vm/ObjectGroup.h	Sat Mar 06 01:05:21 2021 +0000
@@ -49,8 +49,10 @@
   /* Prototype shared by objects in this group. */
   GCPtr<TaggedProto> proto_;  // set by constructor
 
-  /* Realm shared by objects in this group. */
-  JS::Realm* realm_;  // set by constructor
+#ifndef JS_64BIT
+  // Temporary padding to respect MinCellSize.
+  uint64_t padding_ = 0;
+#endif
 
   // END OF PROPERTIES
 
@@ -59,17 +61,13 @@
     return offsetof(ObjectGroup, proto_);
   }
 
-  static inline uint32_t offsetOfRealm() {
-    return offsetof(ObjectGroup, realm_);
-  }
-
   friend class gc::GCRuntime;
 
   // See JSObject::offsetOfGroup() comment.
   friend class js::jit::MacroAssembler;
 
  public:
-  inline ObjectGroup(TaggedProto proto, JS::Realm* realm);
+  inline explicit ObjectGroup(TaggedProto proto);
 
   const GCPtr<TaggedProto>& proto() const { return proto_; }
 
@@ -77,12 +75,6 @@
 
   void setProtoUnchecked(TaggedProto proto);
 
-  JS::Compartment* compartment() const {
-    return JS::GetCompartmentForRealm(realm_);
-  }
-  JS::Compartment* maybeCompartment() const { return compartment(); }
-  JS::Realm* realm() const { return realm_; }
-
   /* Helpers */
 
   void traceChildren(JSTracer* trc);